### ISSUE TITLE
"no data found" error on decoding a valid JPEG image

### ISSUE THREAD

**Issue by Shnatsel** [image-rs/image#1234]
This happens in `image` from git, revision 092db1e21b3e8736711e2cbe1d9b5245613756c1

## Expected

Image decodes successfully

## Actual behaviour

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Decoding(DecodingError { format: Exact(Jpeg), underlying: Some(Format("no data found")) })', src/main.rs:3:18

## Reproduction steps

![00014699](https://user-images.githubusercontent.com/291257/83201443-c483aa80-a145-11ea-97b6-d2ded73cdc12.jpg)
![00014707](https://user-images.githubusercontent.com/291257/83201445-c5b4d780-a145-11ea-9363-49af65514023.jpg)

```rust
fn main() -> std::io::Result<()> {
    let path = std::env::args().skip(1).next().unwrap();
    let image = image::io::Reader::open(path)?.guess_format().decode().unwrap();
    Ok(())
}
```

**Comment by Shnatsel** [comment]
Likely a duplicate of https://github.com/image-rs/jpeg-decoder/issues/96

**Comment by HeroicKatora** [comment]
These wer fixed in `jpeg-decoder` in `v0.1.20`. Others, however, are still failing.
